client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx

```diff
@@ -104,7 +104,7 @@ const ProcessingStep: React.FC< ProcessingStepProps > = function ( props ) {
 			if ( availableFlows[ flow ] ) {
 				availableFlows[ flow ]().then( ( flowExport ) => {
 					if ( flowExport.default.isSignupFlow ) {
-						recordSignupComplete();
+						recordSignupComplete( { ...destinationState } );
```

Comment:
Just roughly thinking. I get the impression that a specific step should not control and record the signup-complete event. I would expect this to be triggered via a flow unmount effect (where we equally observe for some "success" flag) or to actually mark a step as the "signup completion" step (so any step can be used for closing). WDYT?

Reply:
I thought the same as well! But having found the `recordSignupComplete` I doubted that maybe there was some big decision behind this while I was not involved in Stepper work.

Moreover, `useSignUpStartTracking` is in FlowRenderer. Logically, it makes sense to have the signup complete at a framework, and not step, level too. But I wonder if that's overcomplicated. It would be better to open an issue.

Reply:
Yeah, definitely something for a follow-up issue. It does make sense to have this near/in `useSignUpStartTracking`. It will probably lead to more holistic refactor, but we want these things to be perfect! :D